react-ui/src/pages/contest/ContestGrid.jsx

```diff
@@ -0,0 +1,307 @@
+  VotingSlider,
+} from '../../components';
+
+import CardImageLink from './CardImageLink';
+import useContestSizing from './useContestSizing';
+
+const useStyles = makeStyles((theme) => ({
+  descriptionIcon: {
+    paddingLeft: 4,
+    top: -4,
+  },
+  disabledVoting: {
+    cursor: 'wait',
+  },
+  entry: {
+    backgroundColor: theme.palette.grey[100],
+    display: 'flex',
+    flexDirection: 'column',
+    height: '100%',
+  },
+  entryHeading: {
+    columnGap: 4,
+    display: 'flex',
+    minHeight: 60,
+    padding: 8,
+  },
+  entryImageContainer: {
+    alignItems: 'center',
+    display: 'flex',
+    flexGrow: 1,
+    justifyContent: 'center',
+  },
+  entryInfo: {
+    display: 'flex',
+    flexGrow: 1,
+    paddingTop: 4,
+  },
+  entryRatings: {
+    display: 'flex',
+    flexDirection: 'column',
+    flexShrink: 0,
+    textAlign: 'end',
+  },
+  myRating: {
+    color: theme.palette.grey[600],
+    display: 'flex',
+    fontStyle: 'italic',
+    justifyContent: 'right',
+  },
+  numberSymbol: {
+    marginRight: 4,
+  },
+  votingSlider: {
+    marginTop: 16,
+  },
+}));
+
+function ContestGrid({
+  selectedCategories,
+  setExperimentDrawer,
+  votingExpired,
+}) {
+  const { contestId } = useParams();
+
+  const apiPath = `/contests/${contestId}`;
+  const { data: contest } = useSwrData(apiPath, false);
+
+  const classes = useStyles();
+  const [{ votingDisabled }, setComponentsState] = useComponentsState();
+  const { gridDisplayWidth, winnerDisplayWidth } = useContestSizing();
+
+  const [{ density = 'default' }] = useSettingsState();
+
+  const votingUnavailable = votingDisabled || votingExpired;
+
+  const getGridVariables = useCallback((fullWidth) => {
+    const xs = 12;
+    let sm = 12;
+    let md = 6;
+    let lg = 4;
+
+    if (fullWidth) {
+      md = 12;
+      lg = 12;
+    } else if (density === 'compact') {
+      sm = 6;
+      md = 4;
+      lg = 3;
+    }
+
+    return {
+      xs,
+      sm,
+      md,
+      lg,
+    };
+  }, [density]);
+
+  const {
+    categories,
+    entries,
+    isContestMode,
+  } = contest;
+
+  if (!entries) {
+    return null;
+  }
+
+  return (
+    <Grid container spacing={density === 'compact' ? 1 : 2}>
+      {entries
+        .filter(
+          ({ category }) => !selectedCategories.length
+            || selectedCategories.includes(category),
+        )
+        .map(
+          ({
+            average,
+            category,
+            categoryRank,
+            id,
+            imagePath,
+            imgurId,
+            height,
+            name: entryName,
+            rank,
+            rating,
+            user,
+            width,
+          }) => (
+            // eslint-disable-next-line react/jsx-props-no-spreading
+            <Grid key={id} item {...getGridVariables(rank === '1')}>
+              <Card id={id} className={classes.entry} sx={{ maxWidth: rank === '1' ? winnerDisplayWidth : gridDisplayWidth }}>
+                <TemporarySolution {...{
```

Comment:
TemporarySolution?

Reply:
Ya, for some reason the `CardContent` is causing 150-200ms extra rendering time on `ContestGrid`. This solution fixes it, but I don't know why. I named it temporary to highlight that at some point we may need to find a better, more formal solution.

Reply:
Could you rename it to something like `CardContentWrapper` and add a comment why it's there instead?

Feels strange having a component named `Temporary`. I've often found that temporary solutions often end up becoming permanent, so would prefer the reason for it to be captured in a comment.